src/drive/web/modules/views/OnlyOffice/OnlyOfficeProvider.jsx

```diff
@@ -30,13 +34,45 @@ const OnlyOfficeProvider = ({
   const [editorMode, setEditorMode] = useState(
     officeDefaultMode(isDesktop, isMobile)
   )
+
+  const [hasFileDiverged, setFileDiverged] = useState(false)
+  const [officeKey, setOfficeKey] = useState(null)
+
   const isEditorModeView = useMemo(() => editorMode === 'view', [editorMode])
 
   const isFromCreate = useMemo(
     () => pathname.endsWith('/fromCreate'),
     [pathname]
   )
 
+  const fileQuery = buildFileByIdQuery(fileId)
+  const fileResult = useQuery(fileQuery.definition, fileQuery.options)
+
+  const handleFileUpdated = useCallback(
+    data => {
+      if (
+        fileResult?.data?.md5sum !== data.md5sum &&
+        data.cozyMetadata.uploadedBy.slug !== 'onlyoffice-server'
+      ) {
+        setFileDiverged(true)
+      }
+    },
+    [fileResult?.data?.md5sum]
+  )
```

Comment:
You should definitively add a comment to explain why we check the md5sum and the uploadedBy. this is clear today as we work on it, later in a few months, we'll have forgotten. So maybe having a long paragraph explaining our solution can be useful.

Reply:
Its done